lib/semian/protected_resource.rb

```diff
@@ -2,35 +2,59 @@ module Semian
   class ProtectedResource
     extend Forwardable
 
-    def_delegators :@resource, :destroy, :count, :semid, :tickets, :name
+    def_delegators :@bulkhead, :destroy, :count, :semid, :tickets, :name
```

Comment:
@sirupsen the public interface has not changed as much as I am aware - the delegator methods are still the same and `@resource` was never accessible. I can alias resource to bulkhead just not sure its needed.